ArtOfIllusion/src/artofillusion/LayoutWindow.java

```diff
@@ -2184,18 +2225,46 @@ public void transformObjectCommand()
         scaledObjects.add(obj);
       }
     }
-    for (i = 0; i < sel.length; i++)
+    for (int i = 0; i < sel.length; i++)
     {
-      info = theScene.getObject(sel[i]);
-      theScene.objectModified(info.getObject());
+        theScene.objectModified(theScene.getObject(sel[i]).getObject());
     }
-    ArrayList<ObjectInfo> modified = new ArrayList<ObjectInfo>();
+    final ArrayList<ObjectInfo> modified = new ArrayList<ObjectInfo>();
     for (int index : sel)
       modified.add(theScene.getObject(index));
     theScene.applyTracksAfterModification(modified);
     setUndoRecord(undo);
-    updateImage();
   }
+    /** Determine a rotation matrix given the appropriate angles
+     */
+        
+    public static Mat4 determineRotationMatrix(double x, double y, double z) {
+        Mat4 m = Mat4.identity();
+        if (!Double.isNaN(x))
+            m = m.times(Mat4.xrotation(x*Math.PI/180.0));
+        if (!Double.isNaN(y))
+            m = m.times(Mat4.yrotation(y*Math.PI/180.0));
+        if (!Double.isNaN(z))
+            m = m.times(Mat4.zrotation(z*Math.PI/180.0));
+        return m;
+    }
```

Comment:
Does this double up on some of the options available in the `artofillusion.math` package? See `Mat4` and `CoordinateSystem`

If not, should this option be over there instead?